tern/analyze/default/container/multi_layer.py

```diff
@@ -70,20 +70,32 @@ def fresh_analysis(image_obj, curr_layer, shell, driver, envs):
 
 
 def analyze_subsequent_layers(image_obj, shell, master_list, redo,
-                              driver=None):
+                              layer=None, layer_inclusive=False, driver=None):
```

Comment:
I don't think we want/need to pass `layer_inclusive` to the analysis portion of this. Tern analyzes each layer in context with the previous layer, so skipping the previous layers will provide inaccurate results about the packages that exist in a given layer. The `layer_inclusive` flag is really only important for the actual printing of the report to know if we should *print* the layers leading up to `print_layer` or not. As far as analysis goes, however, we want to perform the analysis up until the `print_layer` no matter the value of the `print_inclusive` flag.

What I mean by "Tern analyzes each layer in context with the previous layer" is this.. say we have the following container and the user runs `tern report -i <image> -layer 3`:

```
Base layer = base OS packages
2nd layer = install package set A
3rd layer = install package set B
4th layer = binary app
```

We will analyze the base layer no matter what, but if we don't analyze the second layer and skip straight to the third layer (which is what the user wants information from) then Tern will find package set A AND package set B and report them as having been found in layer 3 since we did not analyze layer 3 in context with layer 2. 

Tern works this way because containers share resources (i.e. packages) across layers. Therefore, as we analyze a layer, we only want to find *unique* packages in that layer despite all the packages from the previous layers being visible in the layer. So in the example above, if we're looking at the third layer, we only want to recognize package set B as existing in layer 3 despite the base OS packages and package set A being detectable in the layer. Check out the `update_master_list() function in `tern/analyze/default/default_common.py` to see how this is done.

Reply:
@aalexanderr let me know if you have any questions about this. Happy to chat on slack too if that's easier.

Reply:
Your comment covers it in great detail, thank you for the clarification. I had some trouble with const & action addition to argparse.add_argument and had to switch over to something else for a bit, yet I dont want to block any other tern work- how quickly do you need this PR fixed?

Reply:
@aalexanderr Targeting this for our March release but if you want to open a new PR with just your original commits addressing #845 and #846 I can take over the analysis work and you could help review + test it? Or I can wait, whatever you prefer.